routines/SAMIHOM4.m

```diff
@@ -143,11 +143,217 @@
+ . . d GETHOME^SAMIHOM3(.SAMIRESULT,.SAMIARG) ; on error go home 
+ . n root s root=$$setroot^%wd("vapals-patients")
+ . s vars("saminame")=$g(@root@(sien,"saminame"))
+ . s vars("dob")=$g(@root@(sien,"dob"))
+ . s vars("sbdob")=$g(@root@(sien,"dob"))
+ . s vars("gender")=$g(@root@(sien,"gender"))
+ . s vars("sex")=$g(@root@(sien,"sex"))
+ . s vars("icn")=$g(@root@(sien,"icn"))
+ . s vars("ssn")=$g(@root@(sien,"ssn"))
+ . s vars("last5")=$g(@root@(sien,"last5"))
+ . s vars("dfn")=$g(@root@(sien,"dfn"))
+ . m SAMIARG=vars
+ . n form,err,zhtml
+ . s form="vapals:addperson"
+ . d RTNPAGE^SAMIHOM4(.SAMIRESULT,form,.SAMIARG) q  ;
+ ;
+ i route="register" d  q  ;
+ . m SAMIARG=vars
+ . d REG^SAMIHOM4(.SAMIRESULT,.SAMIARG)
+ ; 
  quit  ; End of WSVAPALS
  ;
  ;
+REG(SAMIRTN,SAMIARG) ; manual registration
+ ;
+ m ^gpl("reg")=SAMIARG
+ n ssn s ssn=SAMIARG("ssn")
+ s ssn=$tr(ssn,"-")
+ s SAMIARG("errorMessage")=""
+ s SAMIARG("errorField")=""
+ ; test for duplicate ssn
+ ;
+ i $$DUPSSN(ssn) d  ;
+ . ;s SAMIARG("errorMessage")=SAMIARG("errorMessage")_" Duplicate SSN."
+ . s SAMIARG("errorMessage")=SAMIARG("errorMessage")_" Duplicate SSN error. A person with that SSN is already entered in the system."
+ . s SAMIARG("errorField")="ssn"
+ ;
+ ; test for duplicate icn
+ ;
+ n icn s icn=$g(SAMIARG("icn"))
+ i $$DUPICN(icn) d  ;
+ . s SAMIARG("errorMessage")=SAMIARG("errorMessage")_" Duplicate ICN error. A person with that ICN is already entered in the system."
+ . s SAMIARG("errorField")="icn"
+ ;
+ ; test for wellformed ICN
+ ;
+ i $$BADICN(icn) d  ;
+ . s SAMIARG("errorMessage")=SAMIARG("errorMessage")_" Invalid ICN error. The check digits in the ICN do not match"
+ . s SAMIARG("errorField")="icn"
+ ;
+ ; if there is an error, send back to edit with error message
+ i $g(SAMIARG("errorMessage"))'="" d  q  ;
+ . n form
+ . s form="vapals:addperson"
+ . d RTNERR(.SAMIRESULT,form,.SAMIARG)
+ ;
+ n root s root=$$setroot^%wd("patient-lookup")
+ n proot s proot=$$setroot^%wd("vapals-patients")
+ n ptlkien s ptlkien=""
+ n sid s sid=$g(SAMIARG("studyid"))
+ n dfn s dfn=""
+ n sien s sien=""
+ i sid'="" d  ; this is an edit
+ . s sien=$$SID2NUM^SAMIHOM3(sid)
+ . s dfn=$g(@proot@(sien,"dfn"))
+ . s pklkien=$o(@root@("dfn",dfn,"")) ; patient lookup ien
+ i ptlkien="" s ptlkien=$o(@root@("AAAAAA"),-1)+1
+ n name s name=$g(SAMIARG("name"))
+ s @root@(ptlkien,"saminame")=name
+ s @root@(ptlkien,"sinamef")=$p(name,",",1)
+ s @root@(ptlkien,"sinamel")=$p(name,",",2)
+ n fmdob s fmdob=$$FMDT^SAMIUR2(SAMIARG("dob"))
+ n ptlkdob s ptlkdob=$$FMTE^XLFDT(fmdob,7)
+ s ptlkdob=$TR(ptlkdob,"/","-")
+ s @root@(ptlkien,"dob")=ptlkdob
+ s @root@(ptlkien,"sbdob")=ptlkdob
+ n gender s gender=SAMIARG("gender")
+ s @root@(ptlkien,"gender")=$s(gender="M":"M^MALE",1:"F^FEMALE")
+ s @root@(ptlkien,"sex")=SAMIARG("gender")
+ s @root@(ptlkien,"icn")=SAMIARG("icn")
+ s @root@(ptlkien,"ssn")=ssn
+ n last5 s last5=$$UCASE($e(name,1))_$e(ssn,6,9)
+ s @root@(ptlkien,"last5")=last5
+ ;n dfn
```

Comment:
Why is this commented?

Reply:
Tagging @glilly to respond

Reply:
because we no longer want to new dfn at this point. At some future time, we may restructure the code and want to new dfn then. gpl